cmd/frontend/graphqlbackend/external_services.go

```diff
@@ -214,8 +215,25 @@ func (r *schemaResolver) DeleteExternalService(ctx context.Context, args *delete
 		return nil, err
 	}
 
-	if err = r.db.ExternalServices().Delete(ctx, id); err != nil {
-		return nil, err
+	if args.Async {
+		// run deletion in the background and return right away
+		go func() {
+			if err := r.deleteExternalService(context.Background(), id, es); err != nil {
+				log15.Warn("Background external service deletion failed", "err", err)
+			}
+		}()
+	} else {
+		if err = r.deleteExternalService(ctx, id, es); err != nil {
+			return nil, err
+		}
+	}
+
+	return &EmptyResponse{}, nil
+}
```

Comment:
Nit: We can reduce the nesting by returning early inside the `if`.

```suggestion
	if args.Async {
		// run deletion in the background and return right away
		go func() {
			if err := r.deleteExternalService(context.Background(), id, es); err != nil {
				log15.Warn("Background external service deletion failed", "err", err)
			}
		}()
		
		return &EmptyResponse{}, nil
	} 
	
	if err = r.deleteExternalService(ctx, id, es); err != nil {
		return nil, err
	}

	return &EmptyResponse{}, nil
}
```

Reply:
Question: Should we return a message to indicate a background job has been accepted to delete the repo when `async` is `true`?

Reply:
I think this nesting if not that bad, it is a basic `if err = nil` thing, which doesn't contain a lot of logic inside nested `if`.

From my POV reading "if async" will naturally lead to question "what if not?" and the answer is in the `else` clause which I'd like to leave as is. Also not introducing an extra duplicated `return` statement is also better IMO

Reply:
> Should we return a message to indicate a background job has been accepted to delete the repo when `async` is `true`?

This requires some frontend work as well, which is not really planned at the moment. Empty response is good for now